src/sp/it/pl/util/dev/Util.kt

```diff
@@ -3,50 +3,13 @@
 
 package sp.it.pl.util.dev
 
-import javafx.application.Platform
 import javafx.beans.value.ObservableValue
 import mu.KotlinLogging
 import sp.it.pl.util.reactive.attach
 import java.io.PrintWriter
 import java.io.StringWriter
-import java.lang.reflect.Field
-import java.lang.reflect.Modifier
 import kotlin.reflect.KClass
 
-fun fail(): Nothing = fail { null }
-
-inline fun fail(message: () -> String?): Nothing = throw AssertionError(message())
-
-fun throwIf(v: Boolean) = throwIf(v, { "" })
-
-inline fun throwIf(v: Boolean, message: () -> String) {
-    if (v) fail { "Requirement condition not met: ${message()}" }
-}
-
-fun throwIfNot(v: Boolean) = throwIf(!v, { "" })
-
-inline fun throwIfNot(v: Boolean, message: () -> String) = throwIf(!v, message)
-
-fun <T> noNull(o: T?) = o.noNull()
-
-inline fun <T> T?.noNull(message: () -> String = { "" }): T = this ?: fail { "Null forbidden: ${message()}" }
-
-fun throwIfFxThread() {
-    throwIf(Platform.isFxApplicationThread()) { "Must not be invoked on FX application thread!" }
-}
-
-fun throwIfNotFxThread() {
-    throwIf(!Platform.isFxApplicationThread()) { "Must be invoked on FX application thread!" }
-}
-
-fun Field.throwIfFinal() = apply {
-    throwIf(Modifier.isFinal(modifiers)) { "Final field forbidden. Field=$declaringClass.$name" }
-}
-
-fun Field.throwIfNotFinal() = apply {
-    throwIf(!Modifier.isFinal(modifiers)) { "Non final field forbidden. Field=$declaringClass.$name" }
-}
-
 /** Prints the time it takes to execute specified block in milliseconds. */
 fun <T> measureTimeMs(block: () -> T): T {
```

Comment:
I propose to rename the file with the remaining stuff into `Debug`. Although perhaps Fail is also bad, how about `FailUtil` and `DebugUtil`? We need to disambiguate normal classes from utils to not clutter namespace and autocomplete and not confuse developer looking for a class/method.